cmd/validators/main.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/iov-one/weave/x/validators"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		printUsage(os.Stderr)
+		os.Exit(2)
+	}
+
+	log.SetOutput(os.Stderr)
+	log.SetFlags(0)
+
+	cmd, ok := commands[os.Args[1]]
+	if !ok {
+		fmt.Fprintf(os.Stderr, "Unknown command: %q\n\n", os.Args[1])
+		printUsage(os.Stderr)
+		os.Exit(2)
+	}
+	// Cut out command from the parameters, so that the flag module parse correctly.
+	os.Args = append([]string{os.Args[0]}, os.Args[2:]...)
+	if err := cmd(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+// printUsage writes this application usage summary.
+func printUsage(w io.Writer) {
+	var cmds []string
+	for name := range commands {
+		cmds = append(cmds, name)
+	}
+	fmt.Fprintf(w, `Usage:
+	%s <cmd> [options]
+
+Use <cmd> -h to display help for each command.
+Available commands: %s
+`, os.Args[0], strings.Join(cmds, ", "))
+}
+
+// commands is a global register of all commands provided by this program. Each
+// command should use flag package to support options and provide help text.
+var commands = map[string]func() error{
+	"list": cmdList,
+	"add":  cmdAdd,
+}
+
+func cmdList() error {
+	var (
+		tmAddrFl = flag.String("tm", "https://bns.NETWORK.iov.one", "Tendermint node address. Use proper NETWORK name.")
+	)
+	flag.Parse()
+
+	info, err := listValidators(*tmAddrFl)
+	if err != nil {
+		return fmt.Errorf("cannot list validators: %s\n", err)
+	}
+	b, err := json.MarshalIndent(info, "", "\t")
+	if err != nil {
+		return fmt.Errorf("cannot serialize: %s", err)
+	}
+	os.Stdout.Write(b)
+	return nil
+}
+
+func listValidators(nodeURL string) ([]*validatorInfo, error) {
+	req, err := http.NewRequest("GET", nodeURL+"/validators", nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create HTTP request: %s", err)
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot do HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := ioutil.ReadAll(io.LimitReader(resp.Body, 1e5))
+		return nil, fmt.Errorf("unexpected response: %d %s", resp.StatusCode, string(b))
+	}
+
+	var payload struct {
+		Result struct {
+			Validators []*validatorInfo
+		}
+	}
+	if err := json.NewDecoder(resp.Body).Decode(&payload); err != nil {
+		return nil, fmt.Errorf("cannot decode response: %s", err)
+	}
+	return payload.Result.Validators, nil
+}
+
+type validatorInfo struct {
+	Address     string     `json:"address"`
+	PubKey      pubKeyInfo `json:"pub_key"`
+	VotingPower string     `json:"voting_power"`
+	Accum       string     `json:"accum"`
+}
+
+type pubKeyInfo struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+func cmdAdd() error {
+	var (
+		tmAddrFl = flag.String("tm", "https://bns.NETWORK.iov.one", "Tendermint node address. Use proper NETWORK name.")
+		pubKeyFl = flag.String("pubkey", "ASrHklGzkWYreMkjmhK9bwqUbDk1+1KflU+wpDAkvZs=", "Base64 encoded, ed25519 public key.")
```

Comment:
pp: this is the public key of the validator to add/ remove. Please add to description so that there is no confusion with the public key of the message signer.